.github/workflows/build-and-test-differential.yaml

```
@@ -76,7 +76,7 @@ jobs:
   clang-tidy-differential:
     needs: build-and-test-differential
     runs-on: ubuntu-22.04
-    container: ghcr.io/autowarefoundation/autoware:universe-devel-cuda
+    container: ghcr.io/autowarefoundation/autoware:core-devel
```

```suggestion
    container: ghcr.io/autowarefoundation/autoware:core-devel-cuda
```

I think we can keep running this on cuda since it will be more inclusive.

In the future we can refactor to simplify this entire workflow.

---

Since the `autoware.core` does not depend on the CUDA library, the `autoware:core-devel-cuda` image no longer exists. Therefore, staying with `autoware:core-devel` is fine.
https://github.com/autowarefoundation/autoware/pull/5363
https://github.com/autowarefoundation/autoware/pkgs/container/autoware/versions